Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -233,8 +234,38 @@ open class ChatChannelListVC: _ViewController,
     public func replaceChannelListController(_ controller: ChatChannelListController) {
         self.controller = controller
         self.controller.delegate = self
-        collectionView.reloadData()
         self.controller.synchronize()
+        collectionView.reloadData()
+    }
+
+    /// The factory method responsible to create the search controller.
+    open func makeSearchController() -> UISearchController? {
+        guard let searchStrategy = components.channelListSearchStrategy else {
+            return nil
+        }
+
+        let resultsController: UIViewController & UISearchResultsUpdating
+        switch searchStrategy {
+        case .messages:
+            let messageSearchVC = components.messageSearchVC.init()
+            messageSearchVC.messageSearchController = controller.client.messageSearchController()
+            messageSearchVC.didSelectMessage = { [weak self] channel, message in
+                self?.router.showChannel(for: channel.cid, at: message.id)
+            }
+            resultsController = messageSearchVC
+        case .channels:
+            let channelSearchVC = components.channelSearchVC.init()
+            channelSearchVC.controller = controller
+            channelSearchVC.didSelectChannel = { [weak self] channel in
+                self?.router.showChannel(for: channel.cid)
+            }
+            resultsController = channelSearchVC
+        }
+
+        let searchController = UISearchController(searchResultsController: resultsController)
+        searchController.searchResultsUpdater = resultsController
+        searchController.searchBar.placeholder = L10n.ChannelList.search
+        return searchController
```

Comment:
Wouldn't it make more sense to have this in the subclass?

Reply:
Which subclass? The search controller is part of the Channel List, and not other entity 🤔